README.md

```diff
@@ -1 +1,159 @@
-# fe-hometask
\ No newline at end of file
+# React & Rockets
+
+**Description:** Challenge for Adjust Frontend Developer
+
+**Owner:** [adjust](https://github.com/adjust)
+
+**Contacts:** frontend-hiring@adjust.com
+
+---
+
+## Welcome to our little coding exercise! 👋
+
+Here you will be given the opportunity to play with JavaScript and Rockets in the same project. For this, we recommend you to block 2-3 hours of your time to complete it.
+
+**Please make sure you have read this page entirely, before starting the challenge**. If you have any doubts, please reach out to us by [opening an issue](#how-to-request-help) as described at the bottom of this page.
+
+**Importantly**, we would like to give all candidates taking this test the same opportunity to solve the exercise in their own way. Because of this, **please do not fork or share this repo (nor your solution) with anybody 🙏🏻**
+
+<img align="center" src="https://i.imgur.com/ekyJNd9.jpg" width="600">
+
+## Instructions
+
+1. Push your solution to a **private repo** in your **personal Github account**
+2. When you are ready for us to take a look, please add [adjust-frontend-hiring][adjust-frontend-hiring] (GitHub user) as collaborator
+
+## General notes
+
+- The final solution must be a working React application.
+  - It's up to you how to bootstrap a new React application.
+- You can use any third-party library if you see a good fit for it.
+- It's not necessary to heavily comment the code, please only leave comments where they are necessary for code comprehension. However, we welcome notes and comments that reflect the higher-level decisions made during the challenge.
+- Aspects to be assessed:
+  - Code Style – meaningful naming conventions, consistent code formatting, and emphasis on code readability and future extensibility.
+  - Quality Assurance – the solution you upload should work as expected, conforming to the specifications of the tasks.
+  - Introducing best practices such as tests, type safety, and linting is appreciated.
+
+Good luck and happy coding! 🚀
+
+---
+
+## Exercise
+
+### TASK #1 - JavaScript
+
+Implement `prepareData` higher order function, which takes an object of filter params `{year, customerName}`, and returns a function that processes a list of missions, by only showing the ones that were launched in `year` and carried a payload belonging to `customerName`.
+
+**Observations:**
+
+- Missions should appear in inverse chronological order (sort), with the exception of those that carried more payloads should appear first.
+- Payloads are carried in the second stage of a rocket and they can belong to multiple customers.
+- It doesn't matter to which `customerName` 'program' each payload belongs to as long as `customerName` is the customer.
+
+**Example:**
+
+Considering we have the list of missions from this [data fixture][data-fixtures], and the following filter params:
```

Comment:
missing the link to the data here

Reply:
Great catch, here's the original fixture. I forgot if we plan to upload the updated specification as a Gist or something? If yes, we should upload the fixture alongside, otherwise I will copy it to this repo too.

Reply:
Do we need so much details for the senior candidates? Maybe examples below are enough?

Answering your question, I think we'll share a repo with them

Reply:
Right, I have updated the example directly referencing the SpaceX API response.